src/Resources/app/administration/src/component/adyen-refund/index.js

```diff
@@ -118,24 +118,10 @@ Component.register('adyen-refund', {
 
             this.allowRefund = this.order.amountTotal > (refundedAmount / 100);
         },
-
-        isAdyenOrder() {
-            const orderTransactions = this.order.transactions;
-            let isAdyen = false;
-            for (let i = 0; i < orderTransactions.length; i++) {
-                if (orderTransactions[i].customFields !== undefined) {
-                    if (orderTransactions[i].customFields.originalPspReference !== undefined) {
-                        isAdyen = true;
-                    }
-                }
-            }
-
-            this.showWidget = isAdyen;
-        }
     },
 
     beforeMount() {
-        this.isAdyenOrder();
+        this.showWidget = this.adyenService.isAdyenOrder(this.order);
```

Comment:
this isAdyenOrder method does not return a boolean as far as I see, can we ensure the type?

Reply:
Added the fix in this [PR](https://github.com/Adyen/adyen-shopware6/pull/518).